lib/ResponsiveReactGridLayout.jsx

```diff
@@ -238,22 +254,41 @@ export default class ResponsiveReactGridLayout extends React.Component<
         cols: newCols
       });
     }
+    const margin = this.getMarginPaddingValue(nextProps.margin, newBreakpoint);
+    const containerPadding = this.getMarginPaddingValue(
+      nextProps.containerPadding,
+      newBreakpoint
+    );
+
     //call onWidthChange on every change of width, not only on breakpoint changes
     this.props.onWidthChange(
       nextProps.width,
-      nextProps.margin,
+      margin,
       newCols,
-      nextProps.containerPadding
+      containerPadding
     );
   }
 
+  /**
+   * Get a value of margin or containerPadding.
+   *
+   * @param  {Array | Object} param Margin | containerPadding, e.g. [10, 10] | {lg: [10, 10], ...}.
+   * @param  {String} breakpoint   Breakpoint: lg, md, sm, xs and etc.
+   * @return {Array}
+   */
+  getMarginPaddingValue = (param: any, breakpoint: string) => {
```

Comment:
I think it'll be better to extract this method into a separate function in this file. It doesn't seem like a member of the class and maybe we should rename it: 'getMarginPaddingValue' -> 'getIndentationValue'